packages/engine/crawlers/sui/collectibles.ts

```diff
@@ -0,0 +1,151 @@
+import type {
+	JsonRpcProvider,
+	PaginatedObjectsResponse,
+	SuiObjectData,
+} from '@mysten/sui.js';
+import { Networks } from '@walless/core';
+import type { CollectibleDocument } from '@walless/store';
+
+export const getSuiCollectibles = async (
+	connection: JsonRpcProvider,
+	owner: string,
+) => {
+	const collectibleDocuments: CollectibleDocument[] = [];
+	let objectResponses: PaginatedObjectsResponse['data'] = [];
+	let nextCursor: PaginatedObjectsResponse['nextCursor'] | undefined;
+	let cursor: PaginatedObjectsResponse['nextCursor'] | undefined;
+	while (nextCursor !== null) {
+		const allOwnedObjects = await connection.getOwnedObjects({
+			owner,
+			cursor,
+			options: {
+				showType: true,
+				showOwner: true,
+				showDisplay: true,
+				showContent: true,
+			},
+		});
+
+		cursor = allOwnedObjects.nextCursor;
+
+		if (allOwnedObjects.hasNextPage && nextCursor !== cursor) {
+			nextCursor = cursor;
+		} else {
+			nextCursor = null;
+		}
+
+		objectResponses = [...objectResponses, ...allOwnedObjects.data];
+	}
+
+	objectResponses
+		.filter((objResponse) => {
+			if (objResponse.data) {
+				return isCollectible(objResponse.data);
+			}
+
+			return false;
+		})
+		.forEach((objResponse) => {
+			if (objResponse.data) {
+				collectibleDocuments.push(
+					getCollectibleDocument(objResponse.data, owner),
+				);
+			}
+		});
+
+	return collectibleDocuments;
+};
+
+const getCollectibleDocument = (
+	objResponse: SuiObjectData,
+	owner: string,
+): CollectibleDocument => {
+	const collectibleId = `${owner}/${objResponse.objectId}`;
+	const formatMetadata = () => {
+		const metadata: CollectibleDocument['metadata'] = {
+			name: 'Unknown',
+			imageUri: 'Unknown',
+			description: 'Unknown',
+			sod: objResponse,
+		};
+		if (
+			objResponse.display?.data &&
+			typeof objResponse.display.data === 'object'
+		) {
+			const objectDisplay = objResponse.display.data;
+			if ('name' in objectDisplay) {
+				metadata.name = objectDisplay.name;
+			}
```

Comment:
It can be `if (objectDisplay.name) { ...}` or `metadata.name = objectDisplay.name || "Unknown"` and dont init at above (just empty object when init metadata)